src-self-hosted/codegen.zig

```diff
@@ -376,6 +376,34 @@ const Function = struct {
         }
     }
 
+    /// Encodes a REX prefix as specified, and appends it to the instruction
+    /// stream. This only modifies the instruction stream if at least one bit
+    /// is set true, which has a few implications:
+    ///
+    /// * The length of the instruction buffer will be modified *if* the
+    /// resulting REX is meaningful, but will remain the same if it is not.
+    /// * Deliberately inserting a "meaningless REX" requires explicit usage of
+    /// 0x40, and cannot be done via this function.
+    fn REX(self: *Function, arg: struct { B: bool = false, W: bool = false, X: bool = false, R: bool = false }) !void {
```

Comment:
Shouldn't this go in `codegen/x86.zig`?

Reply:
That could work. This also is fine. I think it will be more clear how to organize the code once it's a bit more fleshed out.